.github/workflows/ci.yml

```diff
@@ -63,24 +63,18 @@ jobs:
     - name: Verify index migration from 0.3.x to 0.4.x
       run: hack/verify-index-migration.sh
 
-    - name: Create a new release
+    - name: Make release notes
       if: contains(github.ref, 'tags')
-      id: create_release
-      uses: actions/create-release@v1.0.0
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        tag_name: ${{ github.ref }}
-        release_name: Release ${{ github.ref }}
-        draft: false
-        prerelease: false
+      run: |
+        echo 'RELEASE_NOTES<<EOF' >> $GITHUB_ENV
+        TAG=$GITHUB_REF_NAME hack/make-release-notes.sh >> $GITHUB_ENV
+        echo 'EOF' >> $GITHUB_ENV
 
-    - name: Upload binaries to the new release
+    - name: Create a new release
       if: contains(github.ref, 'tags')
-      uses: svenstaro/upload-release-action@v2
+      id: create_release
+      uses: softprops/action-gh-release@v1
       with:
```

Comment:
can we also add `fail_on_unmatched_files: true` (if not already set by default) as a safeguard?